### ISSUE TITLE
Grammerly makes Tab Groups cycle through my custom tab groups

### ISSUE THREAD

**Issue by GerryPeters** [Quicksaver/Tab-Groups#507]
I've been using Tab Groups add-on for as long as it was discontinued as a standard feature in Firefox. . Since I installed Grammerly a few days ago, sometimes when I open Firefox Tab Groups malfunctions. It starts cycling through all the Tab Groups one at a time.

The only fix, is to use task manager to close Firefox and then copy a previous.js into the Firefox folder and change it’s name to sessionstore.js I can then open Firefox. Unfortunately I lose some of the latest history when I have to do this. I’m using Firefox 50.0

Any suggestions,

**Comment by Quicksaver** [comment]
By Grammerly, I assume you mean [Grammarly for Firefox](https://addons.mozilla.org/firefox/addon/grammarly-1/) as I can't find anything by the exact name "Grammerly".

This seems to be the exact same problem as #411. Grammarly is trying to initialize every single tab at startup, even unloaded tabs (which is all of them except for the selected one). With Tab Groups, you can accumulate a large number of tabs, which means Grammarly will take a long time to initialize and cause "Unresponsive script errors", which in turn switch to the tab that triggered it and causes it to switch to that tab's group.

See that issue, in particular the following comment, for more details: https://github.com/Quicksaver/Tab-Groups/issues/411#issuecomment-256310583. I easily reproduced the exact same problem, except with Grammarly instead of with LastPass, with Tab Groups disabled by following the steps I mention there.

In short, Grammarly's startup needs to be optimized to handle a large amount of tabs. There's nothing I can do on my side about that. You should contact its developers about this, as I'm sure they will want to help with optimizing their add-on as best as they can for users with many tabs.

**Comment by GerryPeters** [comment]
Thank you for your quick detailed response. I did submit this to Grammarly. I hope they can fix it
